rwengine/src/data/GameTexts.hpp

```diff
@@ -3,21 +3,42 @@
 #include <string>
 #include <unordered_map>
 
+/**
+ * The game stores strings as 16-bit indexes into the font
+ * texture, which is something simllar to ASCII.
+ */
+using GameString = std::basic_string<char16_t>;
```

Comment:
If this uses something similar to ascii, but not unicode, why exactly do we still want to use char16_t then?
Wouldn't it be wise to have something like a GameCharacter which is a plain uint16_t?

If we have `fromString` anyway to do the conversion anyway, why aren't we working in `std::string` OR GameCharacter then?
All our game strings are loaded from the GXT anyway and the display uses a fixed offset - we never have to assign a meaning (such as unicode) anywhere unless we want to inject our own strings (and for that we assume ascii and convert from out `std::string` for which we'll also assume ascii).
There is a different type of string in GTA which is in the script system - those 8 char strings which are also used for the GXT keys. Those are probably ASCII or, if they aren't case sensitive we can assume them to be 64 bit which have to match. We never have to assign a meaning to those.